agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/context/element/AgentRunInfoContextElement.kt

```diff
@@ -46,3 +46,11 @@ public suspend fun CoroutineContext.getAgentRunInfoElementOrThrow(): AgentRunInf
             "Unable to retrieve AgentRunInfoContextElement from CoroutineContext. " +
                 "Please make sure the AgentRunInfoContextElement is added to the current CoroutineContext."
         )
+
+/**
+ * Retrieves the `AgentRunInfoContextElement` from the current coroutine context, if present.
+ *
+ * @return The `AgentRunInfoContextElement` if it exists in the current coroutine context, or `null` if not found.
+ */
+public suspend fun CoroutineContext.getAgentRunInfoElement(): AgentRunInfoContextElement? =
```

Comment:
`CoroutineContext` receiver is not needed here, it can be safely converted to the regular function. Also, now that you have a function returning null, it would be idiomatic to update `getAgentRunInfoOrThrow` to use the current function instead of again fetching the value from coroutine context manually